src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Phone} matches any of the keywords given.
+ */
+public class PhoneContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public PhoneContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getPhone().value, keyword));
```

Comment:
May be possible to make phone matching more strict than contains. Interesting discussion on whether it makes sense for users to search phone numbers by substrings

Reply:
I referenced the original matching for find_name code in AB3. In this case, it allows for find_phone to search for multiple people at once. eg. "find_phone 123 456" will display people with phone number 123 and 456 (2 separate people)

Reply:
As for phone matching, do you possibly be meaning to add the constraints in Phone class here as well? eg at least 3 digits etc.